src/domain/service/noteSettings.ts

```diff
@@ -131,9 +135,46 @@ export default class NoteSettingsService {
    * @returns team members
    */
   public async getTeamByNoteId(noteId: NoteInternalId): Promise<Team> {
-    return await this.teamRepository.getByNoteId(noteId);
+    let team = await this.teamRepository.getTeamByNoteId(noteId);
+    let parentId = await this.shared.note.getParentNoteIdByNoteId(noteId);
+
+    /**
+     * team.length === 1 means that team contains only creator or owner, it means that team is not specified by user
+     * parentId === null means that note has no parent to inherit its team
+     */
+    while (team.length === 1 && parentId !== null) {
+      team = await this.teamRepository.getTeamByNoteId(parentId);
+      parentId = await this.shared.note.getParentNoteIdByNoteId(parentId);
+    }
+
+    return team;
   }
 
+  /**
+   *
+   * @param noteId - note internal id
+   */
+  public async getTeamMembersByNoteId(noteId: NoteInternalId): Promise<Team> {
```

Comment:
this method duplicates `getTeamByNoteId`. Why it has been added?

Reply:
it is needed because it call different method in storage level (this method returns team with users as objects) it is used in noteSettings router

Reply:
seems messy, you can unify the approach